assets/src/components/resizable-box/index.js

```diff
@@ -34,13 +46,84 @@ export default ( props ) => {
 				bottom: true,
 				left: false,
 			} }
-			onResizeStop={ ( event, direction, elt, delta ) => {
+			onResizeStop={ ( event, direction ) => {
+				const { deltaW, deltaH } = getDelta( event, angle, lastSeenX, lastSeenY, direction );
 				onResizeStop( {
-					width: parseInt( width + delta.width, 10 ),
-					height: parseInt( height + delta.height, 10 ),
+					width: parseInt( width + deltaW, 10 ),
+					height: parseInt( height + deltaH, 10 ),
+					positionTop: parseInt( blockElement.style.top, 10 ),
+					positionLeft: parseInt( blockElement.style.left, 10 ),
 				} );
 			} }
-			onResizeStart={ onResizeStart }
+			onResizeStart={ ( event, direction, element ) => {
+				lastSeenX = event.clientX;
+				lastSeenY = event.clientY;
+				blockElement = element.closest( '.wp-block' );
+				blockElementTop = blockElement.style.top;
+				blockElementLeft = blockElement.style.left;
+				onResizeStart();
+			} }
+			onResize={ ( event, direction, element ) => {
+				const { deltaW, deltaH } = getDelta( event, angle, lastSeenX, lastSeenY, direction );
+
+				const appliedWidth = minWidth <= width + deltaW ? width + deltaW : minWidth;
+				const appliedHeight = minHeight <= height + deltaH ? height + deltaH : minHeight;
+
+				if ( angle ) {
+					//Convert angle from degrees to radians
+					const radianAngle = angle * Math.PI / 180;
+
+					// Compare position between the initial and after resizing.
+					const initialPosition = getBlockPositioning( width, height, radianAngle );
+					const resizedPosition = getBlockPositioning( appliedWidth, appliedHeight, radianAngle );
+					const diff = {
+						left: resizedPosition.left - initialPosition.left,
+						top: resizedPosition.top - initialPosition.top,
+					};
+					// Get new position based on the difference.
+					const originalPos = {
+						left: getPixelsFromPercentage( 'x', parseInt( blockElementLeft, 10 ) ),
+						top: getPixelsFromPercentage( 'y', parseInt( blockElementTop, 10 ) ),
+					};
+					const updatedPos = {
+						left: originalPos.left - diff.left,
+						top: originalPos.top + diff.top,
+					};
+
+					blockElement.style.left = getPercentageFromPixels( 'x', updatedPos.left ) + '%';
+					blockElement.style.top = getPercentageFromPixels( 'y', updatedPos.top ) + '%';
+				}
+
+				/*if ( 0 && angle ) {
+					switch ( direction ) {
```

Comment:
@swissspidy If you have time, would you mind taking a look at this logic to see if you see if something seems to be off or if you have a better idea, perhaps you have a "fresher" eye. I was trying different options and for some reason, the calcuations didn't work as expected, not sure what I was missing. Even brought out a pencil and paper and everything seemed to make sense on paper, ha! I left in the previous tries and just commented it out to check again tomorrow.

Anyway, at this moment the code above kind of works, not smoothly though.